src/huggingface_hub/inference/_client.py

```diff
@@ -1309,3 +1310,50 @@ def _resolve_url(self, model: Optional[str] = None, task: Optional[str] = None)
             # Otherwise, we use the default endpoint
             else f"{INFERENCE_ENDPOINT}/models/{model}"
         )
+
+    def get_model_status(self, model: Optional[str] = None, *, token: Optional[str] = None) -> ModelStatus:
+        """
+            A function which returns the status of a specific model, from the Inference API.
+
+            Args:
+                model (`str`, *optional*):
+                    Identifier of the model for witch the status gonna be returned(retrieve). If model is not provided, 
+                    the model associated with this instance of InferenceClient will be used. The 
+                    identifier should not be a URL.
```

Comment:
```suggestion
                    Identifier of the model for witch the status gonna be checked. If model is not provided, 
                    the model associated with this instance of [`InferenceClient`] will be used. Only InferenceAPI service can be checked so the 
                    identifier cannot be a URL.
```

Reply:
Hello. 

Thank you the you accepted my contribute. I will make the commits now, of your changes. :smile: 

I wrote some tests but I don't know if they are ok. Please let me know what I should to include and what I need to change.

```python
class TestModelStatus(unittest.TestCase):

    # loadable model test case
    def test_loadable_model(self) -> None:
        client = InferenceClient()
        model_status = client.get_model_status("")
        self.assertEqual(model_status.loaded, False)
        self.assertEqual(model_status.state, "Loadable")
        self.assertEqual(model_status.compute_type, "gpu")
        self.assertEqual(model_status.framework, "text-generation-inference")
    
    # too big model test case
    def test_too_big_model(self) -> None:
        client = InferenceClient()
        model_status = client.get_model_status("")
        self.assertEqual(model_status.loaded, False)
        self.assertEqual(model_status.state, "TooBig")
        self.assertEqual(model_status.compute_type, "cpu")
        self.assertEqual(model_status.framework, "transformers")

    # loaded model test case
    def test_loaded_model(self) -> None:
        client = InferenceClient()
        model_status = client.get_model_status("")
        self.assertEqual(model_status.loaded, False)
        self.assertEqual(model_status.state, "Loaded")
        self.assertEqual(model_status.compute_type, "gpu")
        self.assertEqual(model_status.framework, "text-generation-inference")

    # unknown model test case
    def test_unknown_model(self) -> None:
        client = InferenceClient()
        with self.assertRaises(ValueError):
            client.get_model_status("unknown/model")

    # model as url test case    
    def test_model_as_url(self) -> None:
        client = InferenceClient()
        with self.assertRaises(NotImplementedError):
            client.get_model_status("https://unkown/model")
```